src/server-cm/src/main/java/co/oslc/refimpl/cm/gen/servlet/ServiceProviderCatalogSingleton.java

```diff
@@ -223,7 +223,7 @@ private static SortedSet<URI> getServiceProviderDomains(final ServiceProvider se
     * creation to create unique URI paths for each ServiceProvider. 
     *
     */
-    protected static void initServiceProviders (HttpServletRequest httpServletRequest)
+    protected static synchronized void initServiceProviders (HttpServletRequest httpServletRequest)
```

Comment:
Just to double-check, the call to `initServerProviders `is protected from inside this method `getServiceProviders`.
https://github.com/oslc-op/refimpl/blob/0b3a243ca445ec22424534f9a8a27e4ecc59efbc/src/server-qm/src/main/java/co/oslc/refimpl/qm/gen/servlet/ServiceProviderCatalogSingleton.java#L102

To be consistent, we could have done the protection in the other method that calls `initServiceProviders`; namely `getServiceProviderCatalog`.
https://github.com/oslc-op/refimpl/blob/0b3a243ca445ec22424534f9a8a27e4ecc59efbc/src/server-qm/src/main/java/co/oslc/refimpl/qm/gen/servlet/ServiceProviderCatalogSingleton.java#L96

It makes more sense to do it as you suggest here.
But should we remove the synchronized protection from `getServiceProviders`?

Reply:
Thanks for a check around the codebase! I think I hit the unprotected access in         initServiceProviders(httpServletRequest);.

I think it makes sense to protect the whole method instead of synchronization on an object. Synchronization on an object could be a bit faster but has many gotchas to get right.

Reply:
Does your argument mean we should remove the protection from inside the getServiceProviders method? Just to be consistent?

Reply:
Yes

Reply:
Ok. Then I suggest we introduce this change through a LyoDesigner fix.,I can then introduce the change across all 4 servers directly.
OK?

Reply:
Great, thanks!